test/mapping_quality/tiny_loop.mlir

```diff
@@ -0,0 +1,134 @@
+
+// RUN: mlir-neura-opt %t-llvm.mlir \
+// RUN: --assign-accelerator \
+// RUN: --lower-arith-to-neura \
+// RUN: --lower-memref-to-neura \
+// RUN: --lower-builtin-to-neura \
+// RUN: --lower-llvm-to-neura \
+// RUN: --canonicalize-cast \
+// RUN: --canonicalize-live-in \
+// RUN: --leverage-predicated-value \
+// RUN: --transform-ctrl-to-data-flow \
+// RUN: --fuse-control-flow \
+// RUN: --fold-constant \
+// RUN: --insert-data-mov \
+// RUN: --map-to-accelerator="mapping-strategy=heuristic mapping-mode=spatial backtrack-config=customized=3,3" \
+// RUN: | FileCheck %s -check-prefix=SPATIAL
+
+// RUN: mlir-neura-opt %t-llvm.mlir \
+// RUN: --assign-accelerator \
+// RUN: --lower-arith-to-neura \
+// RUN: --lower-memref-to-neura \
+// RUN: --lower-builtin-to-neura \
+// RUN: --lower-llvm-to-neura \
+// RUN: --canonicalize-cast \
+// RUN: --canonicalize-live-in \
+// RUN: --leverage-predicated-value \
+// RUN: --transform-ctrl-to-data-flow \
+// RUN: --fuse-control-flow \
+// RUN: --fold-constant \
+// RUN: --insert-data-mov \
+// RUN: --map-to-accelerator="mapping-strategy=heuristic mapping-mode=spatial-temporal backtrack-config=customized=3,4" \
+// RUN: | FileCheck %s -check-prefix=SPATIAL-TEMPORAL
+
+module {
+  func.func @simple_add_loop() -> i64 {
+    %c0 = arith.constant 0 : index
+    %c10 = arith.constant 1 : i64
+    %c1 = arith.constant 1 : index
+    %c16 = arith.constant 16 : index
+
+    %result = scf.for %i = %c0 to %c16 step %c1 iter_args(%acc = %c10) -> (i64) {
+      %sum = arith.addi %acc, %acc : i64
+      scf.yield %sum : i64
+    }
+    return %result : i64
+  }
+}
+
+// CHECK:        func.func @simple_add_loop() -> i64 attributes {accelerator = "neura"} {
+// CHECK-NEXT:     %0 = "neura.constant"() <{predicate = true, value = 16 : i64}> : () -> i64
+// CHECK-NEXT:     %1 = "neura.constant"() <{predicate = true, value = 1 : i64}> : () -> i64
+// CHECK-NEXT:     %2 = "neura.constant"() <{predicate = true, value = 1 : i64}> : () -> i64
+// CHECK-NEXT:     %3 = "neura.constant"() <{predicate = true, value = 0 : i64}> : () -> i64
+// CHECK-NEXT:     neura.br %3, %2, %0 : i64, i64, i64 to ^bb1
+// CHECK-NEXT:   ^bb1(%4: i64, %5: i64, %6: i64):  // 2 preds: ^bb0, ^bb2
+// CHECK-NEXT:     %7 = "neura.icmp"(%4, %6) <{cmpType = "slt"}> : (i64, i64) -> i1
+// CHECK-NEXT:     neura.cond_br %7 : i1 then %5, %4, %1, %0 : i64, i64, i64, i64 to ^bb2 else %5 : i64 to ^bb3
+// CHECK-NEXT:   ^bb2(%8: i64, %9: i64, %10: i64, %11: i64):  // pred: ^bb1
+// CHECK-NEXT:     %12 = "neura.add"(%8, %8) : (i64, i64) -> i64
+// CHECK-NEXT:     %13 = "neura.add"(%9, %10) : (i64, i64) -> i64
+// CHECK-NEXT:     neura.br %13, %12, %11 : i64, i64, i64 to ^bb1
+// CHECK-NEXT:   ^bb3(%14: i64):  // pred: ^bb1
+// CHECK-NEXT:     "neura.return"(%14) : (i64) -> ()
+// CHECK-NEXT:   }
+
+// SPATIAL:        func.func @simple_add_loop() -> i64 attributes {CompiledII = 5 : i32, RecMII = 3 : i32, ResMII = 1 : i32, accelerator = "neura"} {
```

Comment:
Can we put `CompiledII = 5 : i32, RecMII = 3 : i32, ResMII = 1 : i32, accelerator = "neura"` into a attribute bundle called `MappingInfo`, and make it also include `mapping_strategy = <spatial-only, heuristic>`, `x_tiles` and `y_tiles`. (and `CompiledII` -> `compiledII`, `RecMII` -> `recMII`, `ResMII` -> `resMII`)